File: yh_connect/src/yh_connect_sub.cpp
#include "ros/ros.h"

#include "std_msgs/Float64.h"


void msgCallback(const std_msgs::Float64::ConstPtr& msg) 
{
   
    ROS_INFO("%f", msg->data); //msg의 data를 출력한다.
}

int main(int argc, char** argv)
{
    ros::init(argc,argv, "yh_connect_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_topic2",30,msgCallback);

    ros::spin(); 
    
    return 0;
}

File: yh_connect/scripts/yh_connect_sub.py
#!/usr/bin/python
#-*- coding: utf-8 -*-  
                               

import rospy
from std_msgs.msg import Float32


class ConnectSub:
    def __init__(self):
        self.sub = rospy.Subscriber("yh_topic2",Float32,self.msgCallback)

    def msgCallback(self,msg):
        rospy.loginfo(msg.data)
        


if __name__ == "__main__":
    rospy.init_node("yh_connect_sub")
    connect_sub = ConnectSub()
    rospy.spin()

